app/logical/source/extractor/bilibili.rb

```diff
@@ -22,86 +21,190 @@ def image_urls
       memoize def article_image_urls
         urls = []
 
-        urls += article_json.dig("detail", "modules", "module_top", "display", "album", "pics").to_a.pluck("url")
-        urls += article_json.dig("detail", "modules", "module_content", "paragraphs").to_a.select do |paragraph|
+        urls += article_json.dig("modules", "module_top", "display", "album", "pics").to_a.pluck("url")
+        urls += article_json.dig("modules", "module_content", "paragraphs").to_a.select do |paragraph|
           paragraph["para_type"] == 2
-        end.pluck(:pic).pluck(:pics).flatten.pluck(:url)
+        end.pluck("pic").pluck("pics").flatten.pluck("url")
 
-        urls
+        urls.compact.map { |u| Source::URL.parse(u).full_image_url || u }
+      end
+
+      memoize def post_image_urls
+        urls = []
+
+        urls += post_json.dig("modules", "module_dynamic", "major", "opus", "pics").to_a.pluck("url")
+        urls += post_json.dig("modules", "module_dynamic", "desc", "rich_text_nodes").to_a.select do |node|
```

Comment:
Not really sure if applicable here, but the order of extracted images should be the same as they appear in the post